### PR TITLE
fix: distribute dialog

### PR DESCRIPTION
**Pull request by nijoe1**
### Changes and Improvements:
- **Batch Distribution Support**: Implemented support for batch distribution when processing multiple payouts.
- **Code Refactoring**: Cleaned and optimized the code for better maintainability and readability.
- **CSV Export Enhancement**: Fixed the CSV export functionality to correctly include the amount in the relevant columns.
- **CSV Import Fix**: Corrected the CSV import functionality to properly read the amount from the respective column during data import.
- **Contract Integration**: Integrated new smart contracts that emit `projectId` in the `Distribute Event` for better event tracking.
- **Duplicate Payout Prevention**: Added logic to handle and prevent double payouts.
- **Completed Payout Section Improvements**: Displaying the project name in the "Completed Payouts" section for better transparency and tracking.

---

Fixes: PAR-347

---

### CODE REVIEW

File: src/features/distribute/components/Distributions.tsx [hunk 1]
```diff
@@ -80,6 +93,10 @@ export function Distributions() {
   if (distributions.length === 0) {
     return <EmptyState title="No distribution found" />;
   }
+  const final_distributions = data?.distributions ?? [];
```
Comment (carlbarrdahl) [inline src/features/distribute/components/Distributions.tsx:96]: `finalDistributions` for consistency

**Review by carlbarrdahl — DISMISSED** [review]
Looks good. Would be nice with some tests on the calculations, especially in `useBatchDistribute`.

**Comment by nijoe1** [comment]
> Looks good. Would be nice with some tests on the calculations, especially in `useBatchDistribute`.

Any idea how to test this out. Moreover this is not for calculating the distribution this is about splitting the distribution in batches.

**Review by hussedev — APPROVED** [review]
LGTM

**Pull request merged**